tests/test_read_data.py

```diff
@@ -0,0 +1,396 @@
+
+
+@pytest.mark.parametrize(
+    "backend",
+    [
+        "pandas",
+        pytest.param("cudf", marks=pytest.mark.gpu),
+    ],
+)
+def test_read_data_blocksize_add_filename_jsonl(mock_multiple_jsonl_files, backend):
+    df = read_data_blocksize(
+        input_files=mock_multiple_jsonl_files,
+        backend=backend,
+        file_type="jsonl",
+        blocksize="128Mib",
+        add_filename=True,
+        input_meta=None,
+        columns=None,
+    )
+
+    assert "filename" in df.columns
+    file_names = df["filename"].unique().compute()
+    if backend == "cudf":
+        file_names = file_names.to_pandas()
+
+    assert len(file_names) == NUM_FILES
+    assert set(file_names.values) == {
+        f"test_{file_id}.jsonl" for file_id in range(NUM_FILES)
+    }
+
+
+@pytest.mark.parametrize(
+    "backend",
+    [
+        "pandas",
+        pytest.param("cudf", marks=pytest.mark.gpu),
+    ],
+)
+def test_read_data_blocksize_add_filename_parquet(mock_multiple_parquet_files, backend):
+    with pytest.raises(
+        ValueError,
+        match="add_filename and blocksize cannot be set at the same time for parquet files",
+    ):
+        read_data_blocksize(
+            input_files=mock_multiple_parquet_files,
+            backend=backend,
+            file_type="parquet",
+            blocksize="128Mib",
+            add_filename=True,
+            input_meta=None,
+            columns=None,
+        )
+
+
+@pytest.mark.parametrize(
+    "backend,file_type",
+    [
+        pytest.param("cudf", "jsonl", marks=pytest.mark.gpu),
+        pytest.param("cudf", "parquet", marks=pytest.mark.gpu),
+        ("pandas", "jsonl"),
+        pytest.param(
+            "pandas",
+            "parquet",
+            marks=pytest.mark.xfail(
+                reason="filename column inaccessible with pandas backend and parquet"
+            ),
+        ),
+    ],
+)
+def test_read_data_fpp_add_filename(
+    mock_multiple_jsonl_files, mock_multiple_parquet_files, backend, file_type
+):
+    input_files = (
+        mock_multiple_jsonl_files
+        if file_type == "jsonl"
+        else mock_multiple_parquet_files
+    )
+
+    df = read_data_fpp(
+        input_files=input_files,
+        backend=backend,
+        file_type=file_type,
+        files_per_partition=NUM_FILES,
+        add_filename=True,
+        input_meta=None,
+        columns=None,
+    )
+
+    assert set(df.columns) == {"filename", "id", "text"}
+    file_names = df["filename"].unique().compute()
+    if backend == "cudf":
+        file_names = file_names.to_pandas()
+
+    assert len(file_names) == NUM_FILES
+    assert set(file_names.values) == {
+        f"test_{file_id}.{file_type}" for file_id in range(NUM_FILES)
+    }
+
+
+@pytest.mark.parametrize(
+    "backend",
+    [
+        "pandas",
+        pytest.param("cudf", marks=pytest.mark.gpu),
+    ],
+)
+@pytest.mark.parametrize(
+    "file_type,add_filename,function_name",
```

Comment:
Instead of `function_name` I would make this a bool and call it `read_with_blocksize` or `read_with_files_per_partition`.

Reply:
I think in pytest when we use bools, it becomes harder to understand which test has failed. For instance here a test will be read as `cudf-parquet-True-read_data_fpp` , and the bool for `add_filename` is already undescriptive. Renaming even `function_name` to blocksize would be even harder to parse `cudf-parquet-True-True`. It's a small nuance, but I would prefer keeping it the same way. If you feel strongly I can change, but I thought I'll share the intent

Reply:
Fair enough, feel free to keep as is. Thanks.